tasks/discover-granules/tests/index.js

```diff
@@ -259,3 +277,148 @@ test('discover granules using S3 throws error when discovery fails',
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', checkDuplicateRewire);
+      await t.throwsAsync(
+        () => handleDuplicates(t.context.filesByGranuleId, 'error')
+      );
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+test('handleDuplicates throws Error on an invalid duplicateHandling configuration',
+  async (t) => {
+    const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+    await t.throwsAsync(
+      () => handleDuplicates(t.context.filesByGranuleId, 'foobar')
+    );
+  });
+
+test.serial('handleDuplicates does not filter when duplicateHandling is set to "replace" or "version"',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', checkDuplicateRewire);
+      const replaceActual = await handleDuplicates(t.context.filesByGranuleId, 'replace');
+      const versionActual = await handleDuplicates(t.context.filesByGranuleId, 'version');
+      t.deepEqual(replaceActual, t.context.filesByGranuleId);
+      t.deepEqual(versionActual, t.context.filesByGranuleId);
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+
+test.serial('filterDuplicates returns a set of filtered keys',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const filterDuplicates = discoverGranulesRewire.__get__('filterDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', async (key) => {
+        if (key === 'duplicate') {
+          return '';
+        }
+        return key;
+      });
+
+      const actual = await filterDuplicates(['duplicate', 'key1', 'key2'], 'bogusHandlingValue');
+      t.deepEqual(actual, ['key1', 'key2']);
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+test.serial('checkDuplicate returns false when API lambda returns a granule',
+  async (t) => {
+    let granulesRevert;
+    try {
+      const checkDuplicate = discoverGranulesRewire.__get__('checkDuplicate');
+      granulesRevert = discoverGranulesRewire.__set__('granules', {
+        getGranule: async () => ({ statusCode: 200, body: '{}' })
+      });
+      const actual = await checkDuplicate('granuleId', '');
+      t.is(actual, false);
+    } finally {
+      granulesRevert();
+    }
+  });
+
+test.serial('checkDuplicate throws an error when API lambda returns a granule and duplicateHandling is set to "error"',
+  async (t) => {
+    let granulesRevert;
+    try {
+      const checkDuplicate = discoverGranulesRewire.__get__('checkDuplicate');
+      granulesRevert = discoverGranulesRewire.__set__('granules', {
+        getGranule: async () => ({ statusCode: 200, body: '{}' })
+      });
+      await t.throwsAsync(checkDuplicate('granuleId', 'error', { get: async () => 'dummy value' }));
```

Comment:
There is no third argument to `checkDuplicate`, is there?

Reply:
Good catch - there *was* during a previous iteration of this.    👍

Reply:
6d4465d14